peachjam/adapters/base.py

```diff
@@ -41,3 +48,16 @@ def get_edit_url(self, document):
     @classmethod
     def name(cls):
         return cls.__name__
+
+
+class RequestsAdapter(Adapter):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.client = requests.session()
+        self.api_url = self.settings["api_url"]
+
+    def client_get(self, url, **kwargs):
+        logger.debug(f"GET {url} kwargs={kwargs}")
+        r = self.client.get(url, **kwargs)
+        r.raise_for_status()
+        return r
```

Comment:
I wanted to do this when I worked on the deletion adapter, because we use a post method on that one, I wanted a generic client. but this is fine since we don't have a need for then post method generally